### PR TITLE
Taxii 2 feed parse sco objects

### PR DESCRIPTION
**Pull request by GuyAfik**
## Status
- [ ] In Progress
- [x] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: [link to the issue](https://jira-hq.paloaltonetworks.local/browse/CIAC-767)

## Description
Add support for parsing the SCO indicators in Taxii2 Feed

## Must have
- [x] Tests
- [ ] Documentation 

**Note** - this is blocked by this [PR](https://github.com/demisto/content/pull/20447) - adding *ASN* indicator type to XSOAR - merged

### CODE REVIEW

**Review by JudahSchwartz — COMMENTED** [review]
looks good. A couple questions. We're waiting on the other prs before merge, right?

File: Packs/ApiModules/Scripts/TAXII2ApiModule/TAXII2ApiModule.py
Comment (GuyAfik) [reply Packs/ApiModules/Scripts/TAXII2ApiModule/TAXII2ApiModule.py:710]: 3 is the score (Malicious) for an indicator, Dror told me it should be Unknown (0), so i need to update that one. I thought by mistake 3 is unknown, while in-fact 3 is malicious (bad) 
image

File: Packs/ApiModules/Scripts/TAXII2ApiModule/TAXII2ApiModule.py
Comment (GuyAfik) [reply Packs/ApiModules/Scripts/TAXII2ApiModule/TAXII2ApiModule.py:722]: because if you take a look at the `parse_generator_type_envelope` and `parse_dict_envelope` function, you can see that it extends the results that it finds, hence it must be returned as a `list`. 
image

File: Packs/ApiModules/Scripts/TAXII2ApiModule/TAXII2ApiModule.py
Comment (GuyAfik) [reply Packs/ApiModules/Scripts/TAXII2ApiModule/TAXII2ApiModule.py]: fixed the dbot-score to be more clear instead of using a raw number

File: Packs/ApiModules/Scripts/TAXII2ApiModule/TAXII2ApiModule.py
Comment (GuyAfik) [reply Packs/ApiModules/Scripts/TAXII2ApiModule/TAXII2ApiModule.py:722]: also if you take a look at other indicator parsing, you will see they are returned as lists too

**Comment by GuyAfik** [comment]
> looks good. A couple questions. We're waiting on the other prs before merge, right?

only waiting for the `ASN` indicator type to be merged, see the PR in the description above.

**Review by JudahSchwartz — APPROVED** [review]
Great!

**Pull request merged**
